### PR TITLE
Add per-node details to Nodes tab

### PR DESCRIPTION
**Pull request by dereknex**
## Summary
- Show each node’s name, readiness, CPU, and memory in the Nodes tab
- Highlight Not Ready nodes with error styling and a short reason line
- Keep missing node metrics visible as `-` instead of `0`
- Update fixtures, docs, and display-model tests to cover the new rows

## Testing
- `./scripts/swift-quality-gate.sh local` passed
- Unit coverage updated for node parsing, display mapping, and QA fixtures

### CODE REVIEW

File: Kubebar/Views/NodeDetailsView.swift [hunk 1]
```diff
@@ -5,20 +5,30 @@ struct NodeDetailsView: View {
     let display: NodeTabDisplay
 
     var body: some View {
+        VStack(alignment: .leading, spacing: 10) {
+            summary
+
+            if display.unavailableMessage == nil {
+                if display.rows.isEmpty, display.summary.hasPrefix("0/0") || display.summary.hasPrefix("-") {
```
[the remark this answers is not in the document]
Comment (dereknex) [reply Kubebar/Views/NodeDetailsView.swift]: > The logic for determining if the empty message should be shown relies on checking the prefix of the `summary` string.

Addressed: added an explicit `showsEmptyMessage` field to `NodeTabDisplay`, set it from the node count during display mapping, and updated the view so it no longer parses summary text. Also added test coverage for the empty node state.

File: Kubebar/Views/NodeDetailsView.swift [hunk 2]
```diff
@@ -35,3 +45,81 @@ struct NodeDetailsView: View {
...
+            if row.readiness == .notReady {
+                Image(systemName: "exclamationmark.triangle.fill")
+                    .font(.caption2)
+                    .accessibilityHidden(true)
+            }
+
+            Text(row.statusLabel)
+                .font(.caption.weight(.semibold))
+        }
+        .foregroundStyle(row.readiness == .notReady ? Color.red : Color.secondary)
```
[the remark this answers is not in the document]
Comment (dereknex) [reply Kubebar/Views/NodeDetailsView.swift]: > The styling logic for "Not Ready" nodes (padding, background, border, and foreground color) is repeated multiple times using `row.readiness == .notReady`.

Addressed: centralized the row error-state check in `isErrorState` and moved the Not Ready padding, corner radius, and opacity values into local style constants.

File: KubebarCore/Services/KubectlClusterReader.swift [hunk 3]
```diff
@@ -659,12 +724,44 @@ private struct NodeRecord: Decodable, Equatable, Sendable {
...
+                condition.status == "True"
+        }) {
+            return (
+                reason: normalizedNodeText(pressureCondition.reason) ?? pressureCondition.type,
+                message: normalizedNodeText(pressureCondition.message)
+            )
+        }
+
+        return (reason: nil, message: nil)
     }
```
[the remark this answers is not in the document]
Comment (dereknex) [reply KubebarCore/Services/KubectlClusterReader.swift]: > pressure conditions like `DiskPressure` are never displayed, even though they are mentioned in the requirements brainstorm.

Addressed: node issue extraction now surfaces pressure conditions as health issues, and a node with `Ready=True` plus `DiskPressure=True` is displayed as needing attention with the pressure reason/message. Added a regression test for that case and documented the invariant.

File: KubebarCore/Services/KubectlClusterReader.swift [hunk 4]
```diff
@@ -659,12 +724,44 @@ private struct NodeRecord: Decodable, Equatable, Sendable {
     struct Condition: Decodable, Equatable, Sendable {
         let type: String
         let status: String
+        let reason: String?
+        let message: String?
     }
 
+    let metadata: Metadata?
```
[the remark this answers is not in the document]
Comment (dereknex) [reply KubebarCore/Services/KubectlClusterReader.swift]: > Since `metadata.name` is a mandatory field for Kubernetes objects, making `metadata` non-optional would ensure data integrity during decoding and simplify the downstream logic.

Addressed: `NodeRecord` and `NodeMetricsRecord` now require `metadata.name`, the fallback unknown-node naming path was removed, and the test fixtures were updated to use real node names for nodes and metrics.
